R/griis.R

```diff
@@ -0,0 +1,210 @@
+                 "Mongolia",
+                 "Montegero",
+                 "Montenegro",
+                 "Morocco",
+                 "Mozambique",
+                 "Myanmar",
+                 "Nepal",
+                 "New Zealand",
+                 "Niger",
+                 "Nigeria",
+                 "Norway",
+                 "Pakistan",
+                 "Paraguay",
+                 "Peru",
+                 "Poland",
+                 "Portugal",
+                 "Republic of Moldova",
+                 "Romania",
+                 "Rwanda",
+                 "Saint Kitts and Nevis",
+                 "Saint Vicent and the Grenadines",
+                 "Sao Tome and Principe",
+                 "Saudi Arabia",
+                 "Senegal",
+                 "Serbia",
+                 "Seychelles",
+                 "Sierra Leone",
+                 "Singapore",
+                 "Slovakia",
+                 "Somalia",
+                 "South Africa",
+                 "South Sudan",
+                 "Spain",
+                 "Sri Lanka",
+                 "Sudan",
+                 "Suriname",
+                 "Swaziland",
+                 "Switzerland",
+                 "Taiwan",
+                 "Thailand",
+                 "The former Yugoslav Republic of Macedonia",
+                 "Togo",
+                 "Tunisia",
+                 "Turkey",
+                 "Uganda",
+                 "Ukraine",
+                 "United Arab Emirates",
+                 "United Republic of Tanzania",
+                 "Uruguay",
+                 "Venezuela",
+                 "Viet Nam",
+                 "Yemen",
+                 "Zambia",
+                 "Zimbabwe")
+  if(!is.null(country)){
+    if(!country %in% countries){
+      stop(paste("country should be one of", paste(countries, collapse = ", ")))
+    }
+  }
+  #Kinndom is valid
+  if(!is.null(kindom)){
+    if(!kindom %in% c("animalia", "plantae", "fungi",
+                    "protozoa", "chromista", "others")){
+    stop("kindom should be one of animalia, plantae, fungi,
+         protozoa, chromista, others")
+    }
+  }
+  #verification is Yes or NULL
+  if(!is.null(verified)){
+    if(!verified %in% c("Yes")){
+    stop("verified should be Yes or NULL")
+    }
+    verified <- ifelse(verified == "Yes", "1", "")
+  }
+  #impacts is Yes or NULL
+  if(!is.null(impacts)){
+    if(!impacts %in% c("Yes")){
+    stop("impacts should be Yes or NULL")
+    }
+    impacts <- ifelse(impacts == "Yes", "1", "")
+  }
+  #type is terrestrial...
+  if(!is.null(type)){
+    if(!type %in% c("terrestrial", "freshwater", "marine",
+                  "brackish", "host")){
+    stop("type should be one of terrestrial, freshwater,
+         marine, brackish, host")
+    }
+  }
+  #Parse url and extract table
+  if(!is.null(name)){
+    name <- gsub(" ", "%20", name, fixed = TRUE)
+  }
+  url <- paste0(griis_base(), "name=", name, "&impacts=", impacts,
```

Comment:
I'd prefer that we use `httr::GET` here to have better control over the http request, proper failure behavior, etc.